src/routers/cog.py

```diff
@@ -20,8 +20,18 @@
 
 @router.put("/datasets/{dataset_id}/build-cog")
 def create_cog(dataset_id: int, options: Optional[ProcessOptions], token: Annotated[str, Depends(oauth2_scheme)], background_tasks: BackgroundTasks):
-    """
-    
+    """FastAPI process chain to add a cog-calculation task to the processing queue, with monitoring and logging.
+    Verifies the access token, loads the dataset to calculate, creates a TaskPayload and returns the task to start 
+    calculating in the background.
```

Comment:
to be more precise: It adds the task to the background process of FastAPI. The task metadata is returned to inform the user on the frontend about the queue position and estimated wait time